tests/test_selection/test_drop_high_psi_features.py

```diff
@@ -0,0 +1,243 @@
+        }
+    )
+
+    test = DropHighPSIFeatures(
+        split_col=test_column, split_frac=0.5, split_distinct_value=False
+    )
+    results = test.fit_transform(df)
+    assert results.shape[0] > 0
+    assert results.shape[1] > 0
+
+
+def test_calculation_no_split_columns():
+    """Test the split of the dataframe using different type of variables."""
+    df = pd.DataFrame(
+        {
+            "time": [date(2012, 6, it) for it in range(1, 31)],
+            "A": [it for it in range(0, 30)],
+            "B": [1, 2, 3, 4, 5, 6] * 5,
+        }
+    )
+
+    test = DropHighPSIFeatures(split_frac=0.5, split_distinct_value=True)
+    test.fit_transform(df)
+    assert len(test.psi_values_) == 2
+
+
+def test_switch():
+    """Test the functionality to switch the basis."""
+    import pandas as pd
+
+    df_a = pd.DataFrame(
+        {
+            "a": [1.0, 2, 3, 1],
+            "b": [1.0, 2, 3, 4],
+            "c": [1, 2, 3, 4],
+            "d": [1.7, 4.7, 6.6, 7.8],
+        }
+    )
+
+    df_b = pd.DataFrame(
+        {
+            "a": [4.0, 3, 5, 1],
+            "b": [11.0, 1, 2, 4],
+            "c": [4, 2, 2, 4],
+            "d": [4.7, 4.7, 7.6, 7.8],
+        }
+    )
+
+    df_order = pd.concat([df_a, df_b]).reset_index(drop=True)
+    df_reverse = pd.concat([df_b, df_a]).reset_index(drop=True)
+
+    case = DropHighPSIFeatures(
+        split_frac=0.5, bins=5, switch=False, min_pct_empty_buckets=0.001
+    )
+    case.fit(df_order)
+
+    switch_case = DropHighPSIFeatures(
+        split_frac=0.5, bins=5, switch=True, min_pct_empty_buckets=0.001
+    )
+    switch_case.fit(df_reverse)
+
+    assert case.psi_values_ == switch_case.psi_values_
+
+
+def test_split_df_according_to_col():
+
+    df = pd.DataFrame(
+        {
+            "A": [it for it in range(0, 20)],
+            "B": [1, 2, 3, 4] * 5,
+            "time": [date(2019, 1, it + 1) for it in range(20)],
+        }
+    )
+
+    cut_off = DropHighPSIFeatures(
+        split_col="time", split_frac=0.5, bins=5, min_pct_empty_buckets=0.001
+    )
+    psi = cut_off.fit(df).psi_values_
+
+    assert len(psi) == 2
+
+
+def test_value_error_is_raised(df):
+
+    with pytest.raises(ValueError):
+        DropHighPSIFeatures(split_frac=0, bins=5, min_pct_empty_buckets=0.001)
+
+    with pytest.raises(ValueError):
+        DropHighPSIFeatures(split_frac=1, bins=5, min_pct_empty_buckets=0.001)
+
+    with pytest.raises(ValueError):
+        DropHighPSIFeatures(bins=1)
+
+    with pytest.raises(ValueError):
+        DropHighPSIFeatures(threshold=-1)
+
+    with pytest.raises(ValueError):
+        DropHighPSIFeatures(switch=1)
+
+    with pytest.raises(ValueError):
+        DropHighPSIFeatures(strategy="unknown")
+
+
+def test_variable_definition(df):
+
+    select = DropHighPSIFeatures(variables=["var_1", "var_3", "var_5"], split_frac=0.01)
+    transformed_df = select.fit_transform(df)
+
+    assert transformed_df.columns.to_list() == ["var_0", "var_2", "var_4"]
+
+
```

Comment:
Could we include some tests for the following (some of them are there already):

1. that the split column is not assigned to the variables when those are detected automatically
2. when missing_values is raised and user passes df with na, an error is raised
3. test that _observation_frequency_per_bin is working as expected
4. test for splitting on index and on column
5. test error is raised if split_col has na
6. test split based on date, int and float entered by user in cut_off (both index and col)
7. test split based on fraction with or without unique values for floats and strings, would it also work if datetime format?

Reply:
I was definitely planning to add more tests as we extended the possibilities to split the dataframe. I'll have a good look at the functionalities that have not yet been tested and at the comments you put.